### PR TITLE
fix: rename and generalize fScore

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: evaluators/rank/fScoreEvaluator/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# fScoreEvaluator
```
Comment (nan-wang) [inline evaluators/rank/fScoreEvaluator/README.md:1]: What about following the camel case here? The class can be named as `FScoreEvaluator` as well.
```suggestion
# FScoreEvaluator
```

File: evaluators/rank/fScoreEvaluator/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+evaluate_groundtruth.matches.append(expected_match_2)
+
+def print_evaluation_score(resp):
+    for doc in resp.search.docs:
+       print(f' Evaluation {doc.evaluations[0].op_name}: {doc.evaluations[0].value}')
+
+#evaluate time
+f = Flow().add(name='encoder', uses='encoder.yml').\
+        add(name='indexer', uses='indexer.yml').\
+        add(name='fscoreevaluator', uses='!fScoreEvaluator')
```
Comment (nan-wang) [inline evaluators/rank/fScoreEvaluator/README.md:70]:
```suggestion
        add(name='evaluator', uses='!fScoreEvaluator')
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: evaluators/rank/fScoreEvaluator/__init__.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+from typing import Sequence, Any, Optional
+
+from jina.executors.evaluators.rank import BaseRankingEvaluator
+
+
+class fScoreEvaluator(BaseRankingEvaluator):
```
Comment (nan-wang) [inline evaluators/rank/fScoreEvaluator/__init__.py:6]:
```suggestion
class FScoreEvaluator(BaseRankingEvaluator):
```

Comment (nan-wang) [reply evaluators/rank/fScoreEvaluator/__init__.py:6]: A class should start with capital letter.

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: evaluators/rank/FScoreEvaluator/__init__.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    def evaluate(self, actual: Sequence[Any], desired: Sequence[Any], *args, **kwargs) -> float:
+        """"
+        :param actual: the matched document identifiers from the request as matched by jina indexers and rankers
+        :param desired: the expected documents matches
+        :return the evaluation metric value for the request document
+        """
+        if not desired or self.eval_at == 0:
+            """TODO: Agree on a behavior"""
+            return 0.0
```
Comment (nan-wang) [inline evaluators/rank/FScoreEvaluator/__init__.py:28]: Maybe we can raise a warning
```suggestion
            if not desired:
                self.logger.warning('`desired` is empty: ', f'{desired}')
            if self.eval_at == 0:
                self.logger.warning('`eval_at` is invalid: ', f'{self.eval_at}')
            return 0.0
```

File: evaluators/rank/FScoreEvaluator/__init__.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+        actual_at_k = actual[:self.eval_at] if self.eval_at else actual
+        common_count = len(set(actual_at_k).intersection(set(desired)))
+        recall = common_count / len(desired)
+
+        divisor = min(self.eval_at or len(desired), len(desired))
+
+        if divisor != 0.0:
+            precision = common_count / divisor
+        else:
```
Comment (nan-wang) [inline evaluators/rank/FScoreEvaluator/__init__.py:38]:
```suggestion
        else:
            self.logger.warning('Divided by zero, the precison is set to 0')
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: evaluators/rank/FScoreEvaluator/__init__.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
+from typing import Sequence, Any, Optional
+
+from jina.executors.evaluators.rank import BaseRankingEvaluator
+
+
+class FScoreEvaluator(BaseRankingEvaluator):
+    """
+    :class:`FScoreEvaluator` Gives the f score of a search system result. (https://en.wikipedia.org/wiki/F-score)
+    :param eval_at: the point at which precision and recall are computed, if None give, will consider all the input to evaluate
```
Comment (bwanglzu) [inline evaluators/rank/FScoreEvaluator/__init__.py:9]: can we add a blank line between class description and `param` section?

File: evaluators/rank/FScoreEvaluator/__init__.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    :param beta: Parameter to weight differently precision and recall. When beta is 1, the fScore corresponds to the harmonic mean
+        of precision and recall
```
Comment (bwanglzu) [inline evaluators/rank/FScoreEvaluator/__init__.py:11]: please add docstring for `args` and `kwargs` : https://docs.jina.ai/chapters/docstring/docstring.html#args-and-kwargs

**Review by bwanglzu — CHANGES_REQUESTED** [review]
minor changes needed! great work!

File: evaluators/rank/FScoreEvaluator/__init__.py (hunk 7 above)
Comment (JoanFM) [reply evaluators/rank/FScoreEvaluator/__init__.py:12]: Can u recheck?

File: evaluators/rank/FScoreEvaluator/__init__.py (hunk 6 above)
Comment (JoanFM) [reply evaluators/rank/FScoreEvaluator/__init__.py:10]: Can u recheck?

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**